### ISSUE TITLE
Create Semantic Sort Options across all tab scopes. 

### ISSUE THREAD

**Issue by roamye** [project-lux/lux-marklogic#343]
**Problem Description:**
The front end would like to implement new semantic sort options to its sort toggle.
This issue will support the creation of each new sort (semantic).

**Expected Behavior/Solution:**
Semantic sort options will appear on all sort toggles across tab scopes.   There will be confirmation that this does not open up a system vulnerabililty.  

**Requirements:**
Create a sort to allow scope tabs to sort records by the sort options below in both ascending and descending order: 
Objects:
- Categorized As
- Collection
- Encountered At
- Encountered by name
- Influenced by name
- Materials
- Produced At
- Produced by name
- Responsible Unit
- Technique

Works:
- Categorized As
- Created By name
- Published At
- Published By name

People/Orgs:
- Born/Formed At
- Categorized As
- Died/Dissolved At
- Gender
- Nationality
- Occupation

Places:
- Categorized As

Concepts:
- Categorized As

Events:
- Carried Out By
- Categorized As
- Took Place At

**Needed for promotion:**
If an item on the list is not needed, it should be crossed off but not removed.

- [ ] Wireframe/Mockup - Mike
- [ ] Committee discussions - Sarah
- [ ] Feasibility/Team discussion - Sarah
- [ ] Backend requirements - TBD
- [ ] Frontend requirements- TBD
- [ ] Are new regression tests required for QA - Amy
- [ ] Questions
- List of questions for discussions. Answers should be documented within the issue.

**UAT/LUX Examples:**
- See FE issue: https://github.com/project-lux/lux-frontend/issues/80

**Dependencies/Blocks:**
- Blocking: https://github.com/project-lux/lux-frontend/issues/80
- Blocked By: https://github.com/project-lux/lux-marklogic/issues/10
- New ticket for the collections/unit sorts: https://github.com/project-lux/lux-marklogic/issues/390

**Related Github Issues:**
- non-semantic: https://github.com/project-lux/lux-marklogic/issues/342 
- will decrease confusion for sorting: https://github.com/project-lux/lux-frontend/issues/407

**Related links:**

- Sort Order: https://docs.google.com/spreadsheets/d/1c9kqnSqTVSGSlV4FkX0wXRphwTYlbHuApP500QtQzGU/edit?gid=0#gid=0

**Wireframe/Mockup:**
Place wireframe/mockup for the proposed solution at end of ticket.

**Comment by roamye** [comment]
From IT Team Meeting 10/18 ([notes here](https://docs.google.com/document/d/1vch7ljI3Wqan-HDfTQs8Ii3vHQRISnrGS-TSyTilWEo/edit?tab=t.0)):

- Peter is OK to continue moving forward with other sorts that follow the way `Produce By` was created. 
- Will need to Update FAQ to include how these sorts display primary names on its cards but sort by the names in the record.

**Comment by jffcamp** [comment]
Semantic sorting will create an attack vector. e.g. multiple user sessions performing semantic sorting in unison on large results.

**Comment by roamye** [comment]
This will need to go out with: https://github.com/project-lux/lux-frontend/issues/407 & should be on the same deployment.

**Comment by roamye** [comment]
uat 11/6: 
the existing performance test can be used to evaluate this,  #318 and #342 (if needed)

**Comment by prowns** [comment]
Per 11/15 analysis - this is a very low used event, so low risk of adding this.  Performance testing is not required.

**Comment by clarkepeterf** [comment]
I am going to document the names of the sorts here as I add them. For the UI Labels, I changed the labels from the requirements section above that have  "name" at the end to not have "name", because all of these sorts are names. 

### Objects
| UI Label | Sort Name |
|---------| ---------- |
| Categorized As | |
| Collection | | 
| Encountered At | itemEncounterPlaceName |
| Encountered By | itemEncounterAgentName | 
| Influenced By | |
| Materials | |
| Produced At | itemProductionPlaceName |
| Produced By | itemProductionAgentName |
| Responsible Unit | |
| Technique | |

### Works
| UI Label | Sort Name |
|---------| ---------- |
| Categorized As | |
| Created By | |
| Published At | |
| Published By | |

### People/Orgs
| UI Label | Sort Name |
|---------| ---------- |
| Born/Formed At | |
| Categorized As | |
| Died/Dissolved At | |
| Gender | |
| Nationality | |
| Occupation | |

### Places
| UI Label | Sort Name |
|---------| ---------- |
| Categorized As | |

### Concepts
| UI Label | Sort Name |
|---------| ---------- |
| Categorized As | |

### Events
| UI Label | Sort Name |
|---------| ---------- |
| Carried Out By | |
| Categorized As | |
| Took Place At | |

**Comment by clarkepeterf** [comment]
Semantic sorts are also affected by https://github.com/project-lux/lux-marklogic/issues/363 - e.g. lowercase letters will sort after uppercase letters. I'm not sure if Optic will sort by the collation of the index field, or if the order is baked into `op.orderBy()` More research is needed for this

**Comment by clarkepeterf** [comment]
There are a couple of sorts above that are marked as NEEDS MORE WORK. They require either multiple semantic hops and/or additional criteria to filter sort values. This is not part of the original semantic sort logic and will need to be a separate ticket.

**Comment by brent-hartwig** [comment]
> There are a couple of sorts above that are marked as NEEDS MORE WORK. They require either multiple semantic hops and/or additional criteria to filter sort values. This is not part of the original semantic sort logic and will need to be a separate ticket.

Hi, @roamye, do you want to create these additional tickets or leave to Peter?

**Comment by roamye** [comment]
@brent-hartwig - For  @clarkepeterf to create please.

**Comment by clarkepeterf** [comment]
@roamye I have created the follow up ticket for Collection and Responsible Unit here: https://github.com/project-lux/lux-marklogic/issues/390

**Comment by roamye** [comment]
Approved by UAT

**Comment by roamye** [comment]
Looks good in PROD. Closing.
